test/observer.spec.ts

```diff
@@ -413,6 +413,28 @@ describe('DataLayerObserver unit tests', () => {
     ExpectObserver.getInstance().cleanup(observer);
   });
 
+  [
+    { configLevel: 0, expectedLevel: 0 },
+    { configLevel: 1, expectedLevel: 1 },
+    { configLevel: 2, expectedLevel: 2 },
+    { configLevel: 3, expectedLevel: 3 },
+    // Passing undefined shouldn't change the global log level
+    { configLevel: undefined, expectedLevel: () => Logger.getInstance().level },
+  ].forEach((tc) => {
+    it(`sets the global log level to ${tc.expectedLevel} given config value ${tc.configLevel}`, () => {
+      const expectedLevel = typeof tc.expectedLevel === 'function' ? tc.expectedLevel() : tc.expectedLevel;
+
+      const observer = ExpectObserver.getInstance().create({
+        logLevel: tc.configLevel,
+        rules: [],
+      });
+
+      expect(Logger.getInstance().level).to.equal(expectedLevel);
+
+      ExpectObserver.getInstance().cleanup(observer);
+    });
+  });
+
```

Comment:
In these tests, I would also test string values that are numbers, as those work today, and we wouldn't want to break that.  Your code hasn't broke those, but we should add them to the test.  We should also test that a non numeric string value results in undefined.

Reply:
The interesting part about this is we'll need to set both the config and expected variants to strings unless we do a loose equality comparison in the test -- e.g.:

```
    { configLevel: '3', expectedLevel: '3' },
```

The reason for this is that the string value isn't coerced when assigned; rather, it's coerced when compared to determine if we should write a log entry:

```
    if (level <= this.level || level === LogLevel.RECORD)
```

At this time, a string would be compared with a number, and the string would be coerced to a number.

At least that's the behavior I was seeing locally. Please let me know if I'm missing something or if you're seeing something different.

Reply:
@ndonze-fs Could we keep the tests you have above and consider that as validation that the constructor accepts and sets a value?  That ensures we know the value passed from the window is updated appropriately to the Logger.

Related to coercion, a different test could be added to `logger.spec.ts` that verifies numeric and their string equivalents log appropriately.

Thoughts?

Reply:
@van-fs We could definitely assert this behavior in 2 parts:

- The `DataLayerObserver` constructor does the right thing with the config value provided on the `window` object. As you mentioned, this is captured in the current tests aside from the string variants as @irv-fs pointed out
- Given a string log level value, we log appropriately by level (not captured in tests yet)

Testing for the latter is possible, but may be a little clunky since we'll have to assign a string value to the global log level (casting the type to `number` or `LogLevel` so TypeScript accepts it), call the logger, and then check that methods (e.g. on a mock console instance) were or weren't called based on the log level. I'd be okay moving forward with this if it make sense to you all even though it's not ideal.

Longer term, I'd like to lean into the fact that we don't know what's coming in on the window object and have a `readConfig(window)` or similar which does the heavy lifting of validating config values, converting to appropriate types (e.g. string to number as we'd want for log level), and sets appropriate defaults. I don't think we should pursue that under this scope of work, though.

Reply:
@irv-fs How do you feel about merging this work? Let me know if my comments above didn't address your concern or if you're seeing a different behavior.

Reply:
I'm good with it!